From 44670dd958c1609480ae7bbd9669d71a9a1c9c3b Mon Sep 17 00:00:00 2001
From: Paul Gortmaker <paul.gortmaker@windriver.com>
Date: Fri, 3 May 2013 18:37:51 -0400
Subject: [PATCH] uio: fix compile warning for !RT_FULL
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

We see:

drivers/uio/uio.c:447:20: warning: ‘uio_thread_handler’ defined but not used [-Wunused-function]

since the thread isn't used outside of RT_FULL.  Enclose the
function in the same #ifdef that encloses the use case.

Signed-off-by: Paul Gortmaker <paul.gortmaker@windriver.com>

diff --git a/drivers/uio/uio.c b/drivers/uio/uio.c
index 16cb9c0..06089c3 100644
--- a/drivers/uio/uio.c
+++ b/drivers/uio/uio.c
@@ -444,6 +444,7 @@ static irqreturn_t uio_interrupt(int irq, void *dev_id)
 	return ret;
 }
 
+#ifdef CONFIG_PREEMPT_RT_FULL
 static irqreturn_t uio_thread_handler(int irq, void *dev_id)
 {
 	struct uio_device *idev = (struct uio_device *)dev_id;
@@ -451,6 +452,7 @@ static irqreturn_t uio_thread_handler(int irq, void *dev_id)
 	uio_event_notify_async(idev);
 	return IRQ_HANDLED;
 }
+#endif
 
 struct uio_listener {
 	struct uio_device *dev;
-- 
1.8.1.2

